encoding/thrift/internal/observabilitytest/test/testserviceserver/server.go

```diff
@@ -104,3 +107,50 @@ func (h handler) Call(ctx context.Context, body wire.Value) (thrift.Response, er
 
 	return response, err
 }
+
+type Call_NoWireHandler struct{ impl Interface }
+
+func (h Call_NoWireHandler) Handle(ctx context.Context, nwc *thrift.NoWireCall) (thrift.NoWireResponse, error) {
+	var (
+		args test.TestService_Call_Args
+
+		rw stream.ResponseWriter
+
+		err error
+	)
+
+	if nwc.RequestReader != nil {
+
+		rw, err = nwc.RequestReader.ReadRequest(ctx, nwc.EnvelopeType, nwc.Reader, &args)
+
+	} else {
+		err = args.Decode(nwc.StreamReader)
+	}
+
+	if err != nil {
+		return thrift.NoWireResponse{}, yarpcerrors.InvalidArgumentErrorf(
+			"could not decode (via no wire) Thrift request for service 'TestService' procedure 'Call': %w", err)
+	}
+
+	success, appErr := h.impl.Call(ctx, args.Key)
+
+	hadError := appErr != nil
+	result, err := test.TestService_Call_Helper.WrapResponse(success, appErr)
+	var response thrift.NoWireResponse
+	response.ResponseWriter = rw
```

Comment:
Is there anyway we could handle the ResponseWrite outside of this method? One of the big benefits we could get:
- We would not need `thrift.NoWireResponse`
- We could have a bunch of refactoring and code simplification by using the same piece of code to set the thrift.Response struct.

Reply:
The response write isn't actually happening here.  The object that gets returned understands what to use to perform the write, but doesn't do it; the write happens in `inbound_nowire.go` in the `Handle` method.

We still wouldn't be able to use the same `thrift.Response` struct because its internals are based on ThriftRW's Wire representation (and its respective `envelope.Enveloper` which contains the response body), where the NoWire representation needs a writer as well as specific response body (that implements `stream.Enveloper`).